lib/mozart_fetcher/local_cache.ex

```diff
@@ -1,8 +1,22 @@
 defmodule MozartFetcher.LocalCache do
-  def get_or_store(id, f) do
-    case MozartFetcher.environment() do
-      :prod -> ConCache.get_or_store(:fetcher_cache, id, f)
-      _ -> f.()
+  def get_or_store(id, fun) do
+    ConCache.get(:fetcher_cache, id)
+    |> cond_fetch(id, fun)
+  end
+
+  defp cond_fetch(nil, id, fun) do
+    case fun.() do
+      resp = {:ok, %HTTPoison.Response{status_code: 200}} -> store(id, resp)
+      other -> other
     end
   end
+
+  defp cond_fetch(value, _id, _fun) do
+    value
+  end
+
+  defp store(id, resp) do
+    ConCache.put(:fetcher_cache, id, resp)
```

Comment:
unfortunately `put` returns `:ok`

Reply:
I think this means there is a chance it could return `:error` so I think the _perfect_ approach would be to match on that potential eventuality (maybe just log and metric then return `resp` anyway) but I think this is unlikely and it's fine as is